Assets/Scripts/MovePlayer.cs

```diff
@@ -0,0 +1,237 @@
+        {
+            case directions.Down:
+                playerAnimator.SetTrigger("Move Down");
+                break;
+            case directions.Up:
+                playerAnimator.SetTrigger("Move Up");
+                break;
+            case directions.Left:
+                playerAnimator.SetTrigger("Move Left");
+                break;
+            case directions.Right:
+                playerAnimator.SetTrigger("Move Right");
+                break;
+        }
+    }
+
+    /// <summary>
+    /// Raycasts for collision in the current direction before moving
+    /// </summary>
+    /// <returns>True if there is something in the way, else false.</returns>
+    private bool IsCollisionBlockingMovement(directions attemptedMoveDirection)
+    {
+        RaycastHit2D hit = Physics2D.Raycast(collisionRaycastOrigin.position, normalizedMovementDirection,1f, collisionLayerMask);
+        return hit.collider != null;
+    }
+
+    /// <summary>
+    /// Raycasts for slow terrain in the current direction before moving
+    /// </summary>
+    /// <returns>True if there is slow terrain in the way, else false.</returns>
+    private bool IsEnteringSlowTerrain(directions attemptedMoveDirection)
+    {
+        RaycastHit2D hit = Physics2D.Raycast(collisionRaycastOrigin.position, normalizedMovementDirection, 1f, slowTerrainLayerMask);
+        return hit.collider != null;
+    }
+
+    /// <summary>
+    /// Removes the less influential movement axis and normalizes what's left
+    /// </summary>
+    private void NormalizeMovementVector()
+    {
+        normalizedMovementDirection = movementInput;
+        if (Mathf.Abs(normalizedMovementDirection.x) >= Math.Abs(normalizedMovementDirection.y))
+            normalizedMovementDirection.y = 0;
+        else
+            normalizedMovementDirection.x = 0;
+        normalizedMovementDirection = normalizedMovementDirection.normalized;
+    }
+
+    /// <summary>
+    /// Checks normalized movement and returns a direction used to activate player animations
+    /// </summary>
+    /// <returns></returns>
+    private directions GetDirectionFromNormalizedMovement()
+    {
+        if (normalizedMovementDirection == Vector2.down)
+            return directions.Down;
+        if (normalizedMovementDirection == Vector2.up)
+            return directions.Up;
+        if (normalizedMovementDirection == Vector2.left)
+            return directions.Left;
+        else
+            return directions.Right;
+    }
+
+    /// <summary>
+    /// Move the player in a set interval aligned with the tilemap
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator MovementRoutine()
+    {
+        isMoving = true;
+        NormalizeMovementVector();
+        directions movementDirection = GetDirectionFromNormalizedMovement();
+        if (IsCollisionBlockingMovement(movementDirection))
+        {
+            // TODO Play bump audio
+            PlayAnim(movementDirection, false, true);
```

Comment:
Rather than call PlayAnim in three potential spots, create variables for these parameters and call PlayAnim once at the end of the method once the new variables have meaningful values plugged in.